syntax = "proto3";

service SafeEntry {

    rpc CheckIn (CheckInRequest) returns (CheckInReply) {}
    rpc CheckOut (CheckOutRequest) returns (CheckOutReply) {}
    rpc CheckInGroup (GroupInRequest) returns (CheckInReply) {}
    rpc CheckOutGroup (GroupOutRequest) returns (CheckOutReply) {}

}

message CheckInRequest {
    string name = 1;
    string nric = 2;
    string location = 3;
    string checkin = 4;
}

message CheckInReply {
    string message = 1;
}

message CheckOutRequest {
    string name = 1;
    string nric = 2;
    string checkout = 3;
}

message CheckOutReply {
    string message = 1;
}

message GroupInRequest {
    repeated string name = 1;
    repeated string nric = 2;
    string location = 3;
    string checkin = 4;
}

message GroupOutRequest {
    repeated string name = 1;
    repeated string nric = 2;
    string checkout = 4;
}